src/secp256k1.c

```diff
@@ -413,6 +413,54 @@ static SECP256K1_INLINE void buffer_append(unsigned char *buf, unsigned int *off
     *offset += len;
 }
 
+/* Initializes SHA256 with fixed midstate. This midstate was computed by applying
+ * SHA256 to SHA256("BIPSchnorrDerive")||SHA256("BIPSchnorrDerive"). */
+static void secp256k1_nonce_function_bipschnorr_sha256_tagged(secp256k1_sha256 *sha) {
+    secp256k1_sha256_initialize(sha);
+    sha->s[0] = 0x1cd78ec3ul;
+    sha->s[1] = 0xc4425f87ul;
+    sha->s[2] = 0xb4f1a9f1ul;
+    sha->s[3] = 0xa16abd8dul;
+    sha->s[4] = 0x5a6dea72ul;
+    sha->s[5] = 0xd28469e3ul;
+    sha->s[6] = 0x17119b2eul;
+    sha->s[7] = 0x7bd19a16ul;
+    sha->bytes = 64;
+}
+
+/* This nonce function is described in BIP-schnorr
+ * (https://github.com/sipa/bips/blob/bip-schnorr/bip-schnorr.mediawiki) */
```

Comment:
```suggestion
 * (https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki) */
```

Reply:
404. I'll probably do the name change once the PR is merged. In some cases I'd rather write `BIP-340 ("Schnorr Signatures for secp256k1")`.